### ISSUE TITLE
xargs error

### ISSUE THREAD

**Issue by matthayes** [pre-commit/pre-commit#217]
I followed the main instructions on the website and I keep getting this error below when I commit.  If I separately run "pyvenv . && source ./bin/activate" though and then run "pre-commit run --all-files" they pass.  Sometimes though even this doesn't work and I have to run "pre-commit cleanup" then "pre-commit install" again.  Any advice?

```
Check for added large files...............................................................................................Failed
hookid: check-added-large-files

xargs: check-added-large-files: No such file or directory

Trim Trailing Whitespace..................................................................................................Failed
hookid: trailing-whitespace

xargs: trailing-whitespace-fixer: No such file or directory
```

**Comment by asottile** [comment]
@matthayes Can you paste the following commands for me?

```
$ cat .pre-commit-config.yaml
$ which pre-commit
$ pre-commit --version
$ python -m pre_commit.main --version
$ ls ~/.pre-commit/ -al
```

**Comment by matthayes** [comment]
Here you go:

```
$ cat .pre-commit-config.yaml

-   repo: git@github.com:pre-commit/pre-commit-hooks
    sha: 15b678e9c67c42765c3d579e40672a332f4366d7
    hooks:
    -   id: check-added-large-files
    -   id: trailing-whitespace
```

```
$ which pre-commit

<no output>
```

```
$ pre-commit --version

-bash: pre-commit: command not found
```

```
$ python -m pre_commit.main --version

/usr/bin/python: No module named pre_commit
```

```
$ ls ~/.pre-commit/ -al

db.db		repoy8e8chfa
```

**Comment by matthayes** [comment]
If I run "source ./bin/activate" first and run the same commands:

```
$ which pre-commit

<project dir>/bin/pre-commit
```

```
$ pre-commit --version

pre-commit 0.4.4
```

```
$ python -m pre_commit.main --version

main.py 0.4.4
```

**Comment by matthayes** [comment]
I'd also like to point out something else strange that may be related.  I tried opening a new window and cloning this repo somewhere else just to get a clean state.  Note that the same failure occurs.

```
$ git clone <project dir A> <project dir B>

$ cd <project dir B>

$ pyvenv . && source ./bin/activate

$ pip install -r requirements.txt

$ pre-commit install

pre-commit installed at <project dir>/.git/hooks/pre-commit

$ pre-commit run --all-files

Check for added large files...............................................................................................Failed
hookid: check-added-large-files

xargs: check-added-large-files: No such file or directory

Trim Trailing Whitespace..................................................................................................Failed
hookid: trailing-whitespace

xargs: trailing-whitespace-fixer: No such file or directory

```

When I run `ls bin` in the project directory I do not see check-added-large-files or trailing-whitespace-fixer.

However, if I run the "clean" command and run the hook then they all pass.

```
$ pre-commit clean

$ pre-commit run --all-files

[INFO] Initializing environment for git@github.com:pre-commit/pre-commit-hooks.
[INFO] Installing environment for git@github.com:pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Check for added large files...............................................................................................Passed
Trim Trailing Whitespace..................................................................................................Passed
```

Now when I run "ls bin" I see check-added-large-files or trailing-whitespace-fixer.  It doesn't seem like running "pre-commit clean" should be necessary for this to work.

**Comment by asottile** [comment]
That is quite strange!  pre-commit installs the hook environments into the home directory usually.  What operating system are you using (and where can I get a copy of `pyenv`)?

Also what does:

`bin/python -c "import os.path; print(os.path.expanduser('~'))"`

Give you?

Sorry if this message is derpy I'm on my phone

**Comment by matthayes** [comment]
I'm using OS X 10.10.2.  I think the pyvenv tool was already installed.  It might just come with Python.  The system Python is 2.7.6.  Also note that it is pyvenv -- not pyenv.  

That command just gives my user home directory: /Users/<username>

With pyvenv the various packages get downloaded under a lib directory in the local repository.  I tried adding additional logging to the pre-commit source code under there but I wasn't able to figure out the problem.

**Comment by asottile** [comment]
I'll look into this when I'm at a computer.  I imagine this might be a pyvenv vs. virtualenv difference or something with PYTHONPATH etc.

**Comment by matthayes** [comment]
One other thing to mention is that I have Python 3.4.3 installed as well.  I installed this with `brew install python3`.  I imagine this is where pyvenv is coming from because I think pyvenv was introduced in Python 3.3.  When I look at the pyvenv.cfg file that is generated after the pyvenv command I see it was created from Python version 3.4.3.

**Comment by asottile** [comment]
I'll check this on my work laptop (OSX).  I can't reproduce this using pyvenv on ubuntu unfortunately :/

**Comment by matthayes** [comment]
Okay thanks, let me know if I can help.

**Comment by asottile** [comment]
Oops, busy day at work and didn't get to this.  A couple things to try though: what happens if installed with `virtualenv` instead of pyvenv? Inside of pyvenv what does `which pip` give you.  I also wonder if there's something odd with your environment,  of you're willing to pastebin the output of `env`.

**Comment by asottile** [comment]
I am able to reproduce this on osx.  Lemme see if I can make a small reproducible testcase.  I imagine that either pyvenv or virtualenv is doing the wrong thing for some reason :(

**Comment by asottile** [comment]
oops didn't mean to push that directly to master.  @matthayes can you try current master and see if it fixes your issues? `pip install git+git://github.com/pre-commit/pre-commit` instead of `pip install pre-commit`.  If it fixes it I'll release another version including this fix

**Comment by matthayes** [comment]
When I run `pre-commit run --all-files` after installing from the git repo I'm getting this error below.  Do you know how to fix this?

```
pkg_resources.DistributionNotFound: The 'pep8<1.6,>=1.5.7' distribution was not found and is required by flake8
```

**Comment by asottile** [comment]
This was fixed in version v0.4.1 of pre-commit-hooks.  You can fix this automatically by running `pre-commit autoupdate`

**Comment by asottile** [comment]
I'm going to mark this as fixed, I've gotten a local reproduction that doesn't happen after my patch.  I'll roll a new version soon.
